src/main/java/ch/uzh/soprafs22/groupmatcher/service/MatcherService.java

```diff
@@ -92,4 +91,42 @@ public void createTeams(Long matcherId) {
                 .forEach(teamEntry -> formTeam(matcher, teamEntry));
     }
 
+    public double[][] createAnswerMatrixAll(Long matcherId) {
+
```

Comment:
It might be simpler to to create the matrix in this order:

1. Fetch the matcher from the repository - matcherRepository.findById(matcherId).orElseThrow(...)
2. Use the matcher's getters to create a flat list of the IDs of all answers to all questions:
`matcher.getQuestions().stream().flatMap(question -> question.getAnswers().stream().map(Answer::getId)).toList()`
3. Iterate over the students via  matcher.getStudents() and for each student, map the list of answer IDs from above to a number - the number is 0 if the answer ID is not in the selectedAnswers list of the current student, or it is the number of possible answers to the question (= selectedAnswer.getQuestion().getAnswers().size()) if the answer ID is in the student's selectedAnswers

Ultimately thought the approach above is just an optimisation where we don't need as much code or additional methods / repository queries to get the same result. Besides that as far as I can tell the current code produces the same result (the only thing to change is replace the value 1 with the number of possible answers to the question, see below.)

Reply:
Thanks for your feedback :) I will consider it later~ for this time, I just merge it as this commit has no conflict or problem.